core/src/main/java/org/apache/accumulo/core/data/Mutation.java

```diff
@@ -853,6 +853,475 @@ public void putDelete(byte[] columnFamily, byte[] columnQualifier,
+
+    VisibilityOptions qualifier(Text colQual);
+  }
+
+  /**
+   * Provides methods for setting the column visibility of a Mutation. The user can provide the
+   * visibility as a byte array or {@link org.apache.accumulo.core.security.ColumnVisibility} object
+   * instance and the backend will do the necessary transformation.
+   *
+   * All QualifierOptions methods return an instance derived from the VisibilityOptions interface,
+   * allowing the methods to be semantically chained.
+   *
+   * @since 2.0.0
+   */
+  public interface VisibilityOptions extends TimestampOptions {
+    TimestampOptions visibility(byte[] colVis);
+
+    TimestampOptions visibility(ByteBuffer colVis);
+
+    TimestampOptions visibility(CharSequence colVis);
+
+    TimestampOptions visibility(ColumnVisibility colVis);
+
+    TimestampOptions visibility(Text colVis);
+  }
+
+  /**
+   * Provides methods for setting the timestamp of a Mutation. The user must provide the timestamp
+   * as a long.
+   *
+   * All TimestampOptions methods return an instance derived from the MutationOptions interface,
+   * allowing the methods to be semantically chained.
+   *
+   * @since 2.0.0
+   */
+  public interface TimestampOptions extends MutationOptions {
+    MutationOptions timestamp(long ts);
+  }
+
+  /**
+   * Provides methods for setting the value of a Mutation. The user can provide the value as a byte
+   * array, Value, or ByteBuffer object instance and the backend will do the necessary
+   * transformation.
+   *
+   * All MutationOptions methods complete a fluent Mutation API method chain.
+   *
+   * @since 2.0.0
+   */
+  public interface MutationOptions {
+    Mutation put(byte[] val);
+
+    Mutation put(ByteBuffer val);
+
+    Mutation put(CharSequence val);
+
+    Mutation put(Text val);
+
+    Mutation put(Value val);
+
+    Mutation delete();
+  }
+
+  /**
+   * Fluent API for putting or deleting to a Mutation that makes it easy use different types
+   * (i.e byte[], CharSequence, etc) when specifying the family, qualifier, value, etc.
+   *
+   * <p>
+   * Methods are optional but must follow this order: family, qualifier, visibility, timestamp.
+   *
+   * <p>
+   * The put and delete methods end the chain and finalize the Mutation by filling Mutation's
+   * UnsynchronizedBuffer.
+   *
+   * @return a new FamilyOptions object, advancing the method chain
+   * @since 2.0.0
+   */
+  public FamilyOptions at() {
+    return new Options();
+  }
+
+  // private inner class implementing all Options interfaces
+  private class Options implements FamilyOptions {
+    byte[] columnFamily;
+    int columnFamilyLength;
+
+    byte[] columnQualifier;
+    int columnQualifierLength;
+
+    byte[] columnVisibility = null;
+    int columnVisibilityLength;
+
+    boolean hasTs = false;
+    long timestamp;
+
+    private Options() {}
+
+    // methods for changing the column family of a Mutation
+    /**
+     * Modifies the column family of a mutation.
```

Comment:
Could be : `Sets the family to modify.`  I am ok with the current wording, thought this suggestions was slight more accurate.